packages/store/src/stores/scalarWeakMapStore.js

```diff
@@ -58,8 +58,12 @@ export const makeWeakMapStoreMethods = (
     },
 
     addAll: entries => {
-      if (isCopyMap(entries)) {
-        entries = getCopyMapEntries(entries);
+      if (typeof entries[Symbol.iterator] !== 'function') {
+        if (isCopyMap(entries)) {
+          entries = getCopyMapEntries(entries);
+        } else {
+          Fail`provided data source is not iterable`;
```

Comment:
```suggestion
          Fail`provided data source is not iterable: ${entries}`;
```

Reply:
And, having tried it, I find I'm wrong.  Is the SES console smarter than the regular one in this respect?

However, a non-iterable, non-copyMap ends up getting passed to `isCopyMap` and once again choking on the harden test.  If I harden the parameter I'll get sensible feedback, but if not it complains that the thing is not hardened rather than that it's not something usable as a data source (i.e., your first remedial action in debugging is to harden it, only to THEN find out that it's wrong).

Reply:
For the record: Mark concurs that we can use `Object.isFrozen` to preflight `isCopyMap` and get most of the diagnostic benefit.  This I have done.